PhotoApp/MainViewController/Other/OtherViewController.swift

```diff
@@ -32,15 +56,68 @@ class OtherViewController: UIViewController {
         }
 
     }
+    
+    func addPhoto() {
+        let alert = UIAlertController.init(title: "Add photo", message: nil, preferredStyle: UIAlertController.Style.actionSheet)
+        alert.addAction(UIAlertAction.init(title: "Take A Picture", style: UIAlertAction.Style.default, handler: { (_) in
+            self.openCamera()
+        }))
+        alert.addAction(UIAlertAction.init(title: "Choose From Gallery", style: UIAlertAction.Style.default, handler: { (_) in
+            self.openGallery()
+        }))
+        alert.addAction(UIAlertAction.init(title: "Cancel", style: UIAlertAction.Style.cancel, handler: nil))
+
+        self.present(alert, animated: true, completion: nil)
+
+    }
+    
+    func openCamera() {
+        if UIImagePickerController.isSourceTypeAvailable(UIImagePickerController.SourceType.camera) {
+            let status = AVCaptureDevice.authorizationStatus(for: AVMediaType.video)
+            if status == .denied {} else {
+                self.imagePickerController.sourceType = UIImagePickerController.SourceType.camera
+                self.imagePickerController.allowsEditing = false
+                self.present(self.imagePickerController, animated: true, completion: nil)
 
-    /*
-    // MARK: - Navigation
+            }
+        }
+    }
+
+    func openGallery() {
+        let status = PHPhotoLibrary.authorizationStatus()
+        if status == .denied {} else {
```

Comment:
@Kasiama if status == .authorized? :D But what about another cases: .restricted, .notDetermined? In such cases we should provide an option to request this authority status :)

Reply:
fixed